### PR TITLE
client waitfor: Quit right after cancel

### PR DESCRIPTION
**Pull request by sorah**
Previously `waitfor` accepts signal to cancel the taskset, but it continues to
wait taskset's persistence.

this patch changes to quit on 2nd signal. Useful for users who don't want to
wait the result.

@cookpad/rrrspec :eyeglasses:

### CODE REVIEW

**Comment by sorah** [comment]
Amended this patch to quit right after of cancel.

----

Previously `waitfor` accepts signal to cancel the taskset, but it continues to wait taskset's persistence.

This patch changes to quit right after of cancel.

Also:

- showing message to appeal humans that cancel request has accepted.
- quit on 2nd signal, for situation such `waitfor` can't request cancellation.

File: rrrspec-client/lib/rrrspec/client/cli.rb [hunk 1]
```diff
@@ -79,8 +79,17 @@ def waitfor(taskset_id)
         exit(1) unless taskset.exist?
 
         rd, wt = IO.pipe
-        Signal.trap(:TERM) { wt.write("1") }
-        Signal.trap(:INT) { wt.write("1") }
+
+        cancelled = false
+        do_cancel = proc {
+          exit if cancelled
```
Comment (eagletmt) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: [nits] it should exit with non-zero status in this case

Comment (sorah) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: good catch

**Comment by eagletmt** [comment]
looks good :+1:

File: rrrspec-client/lib/rrrspec/client/cli.rb [hunk 2]
```diff
@@ -79,8 +79,17 @@ def waitfor(taskset_id)
         exit(1) unless taskset.exist?
 
         rd, wt = IO.pipe
-        Signal.trap(:TERM) { wt.write("1") }
-        Signal.trap(:INT) { wt.write("1") }
+
+        cancelled = false
+        do_cancel = proc {
+          exit(1) if cancelled
```
Comment (taiki45) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: めっちゃ nits だけどいつか正常終了する系の処理をキャンセルした時って exist status 0 以外が自然な気が個人的にする

**Comment by taiki45** [comment]
あっ よさそう :+1:

**Comment by sorah** [comment]
ハイ。 merging.

File: rrrspec-client/lib/rrrspec/client/cli.rb [hunk 3]
```diff
@@ -89,6 +98,7 @@ def waitfor(taskset_id)
           elsif rs.size != 0
             rs[0].getc
             taskset.cancel
+            break
```
Comment (draftcode) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: I have a doubt on this line. With this break, the process will quit without waiting for the taskset.

Comment (eagletmt) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: Yes, it's the main purpose of this PR.

When `rrrspec waitfor` is interrupted, rrrspec is expected to cancel the taskset just like `rrrspec cancel` does.
`rrrspec cancel` doesn't care about the cancelled taskset.

Comment (draftcode) [inline rrrspec-client/lib/rrrspec/client/cli.rb]: Oh, I saw the initial PR description only. Never mind.

**Pull request merged**